src/Components/Components/src/Routing/RouteEntry.cs

```diff
@@ -1,12 +1,12 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
 using System.Collections.Generic;
 
 namespace Microsoft.AspNetCore.Components.Routing
 {
-    internal class RouteEntry
+    internal readonly struct RouteEntry
```

Comment:
Could you clarify why we benefit from this becoming a `struct`? There's only a bounded number of them and they are created on app initialization, right?

Reply:
Same as above. Since this is shared global state, I wanted to make it clear this was immutable. There isn't any other benefit to doing this.